go/tools/builders/compilepkg.go

```diff
@@ -343,6 +351,15 @@ func compileArchive(
 		if err != nil {
 			return err
 		}
+		if coverMode != "" {
+			// Compile original source files, not coverage instrumented, for nogo
+			_, goSrcsNogo, _, err = cgo2(goenv, goSrcsNogo, cgoSrcsNogo, cSrcs, cxxSrcs, objcSrcs, objcxxSrcs, sSrcs, hSrcs, packagePath, packageName, cc, cppFlags, cFlags, cxxFlags, objcFlags, objcxxFlags, ldFlags, cgoExportHPath)
```

Comment:
This PR shouldn't affect how cgo analyzers operate, because the source files for nogo are also cgo compiled, they are just not coverage instrumented.